backend/danswer/search/pipeline.py

```diff
@@ -74,9 +86,73 @@ def __init__(
         self._relevant_section_indices: list[int] | None = None
 
         # Generates reranked chunks and LLM selections
-        self._postprocessing_generator: Iterator[
-            list[InferenceSection] | list[int]
-        ] | None = None
+        self._postprocessing_generator: (
+            Iterator[list[InferenceSection] | list[int]] | None
+        ) = None
+
+    def evaluate(
```

Comment:
This should be separated out into a different module. We want to not have things tightly coupled with classes like the SearchPipeline if the functionality can be separated out into a separate module. It makes things more reuseable, more testable, more extensible. So in this case, the bulk of the functionality should be in a new module in secondary_llm_flows (see the way chunk_usefulness is used for reference)

Reply:
Another nit comment on this one, since this function should never be directly called outside of the other functions/properties of this class, let's make it an underscored function if you still end up having it. But better yet just offload all this functionality as mentioned above and just have the property relevance_summaries call out to that instead.

Reply:
Makes sense and agree on all frnots